src/nvim/charset.c

```diff
@@ -1598,10 +1609,14 @@ int vim_islower(int c)
   return islower(c);
 }
 
-int vim_isupper(int c)
+/// Check that the character is upper-case
+///
+/// @param  c  character to check
+bool vim_isupper(int c)
+  FUNC_ATTR_CONST FUNC_ATTR_WARN_UNUSED_RESULT FUNC_ATTR_NONNULL_ALL
```

Comment:
Maybe @ZyX-I can correct me on this, but the [GCC docs](https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html#Common-Function-Attributes) make it seem like the `const` attr. shouldn't be used for functions which call non-const functions, which this does (`utf_isupper` for one).